sdk/include/opentelemetry/sdk/metrics/ungrouped_processor.h

```diff
@@ -0,0 +1,238 @@
+          std::dynamic_pointer_cast<sdkmetrics::MinMaxSumCountAggregator<T>>(record_agg);
+
+      auto temp_batch_agg_raw_mmsc  = temp_batch_agg_mmsc.get();
+      auto temp_record_agg_raw_mmsc = temp_record_agg_mmsc.get();
+
+      temp_batch_agg_raw_mmsc->merge(*temp_record_agg_raw_mmsc);
+    }
+    else if (agg_kind == sdkmetrics::AggregatorKind::Gauge)
+    {
+      std::shared_ptr<sdkmetrics::GaugeAggregator<T>> temp_batch_agg_gauge =
+          std::dynamic_pointer_cast<sdkmetrics::GaugeAggregator<T>>(batch_agg);
+
+      std::shared_ptr<sdkmetrics::GaugeAggregator<T>> temp_record_agg_gauge =
+          std::dynamic_pointer_cast<sdkmetrics::GaugeAggregator<T>>(record_agg);
+
+      auto temp_batch_agg_raw_gauge  = temp_batch_agg_gauge.get();
+      auto temp_record_agg_raw_gauge = temp_record_agg_gauge.get();
+
+      temp_batch_agg_raw_gauge->merge(*temp_record_agg_raw_gauge);
+    }
+    else if (agg_kind == sdkmetrics::AggregatorKind::Sketch)
+    {
+      std::shared_ptr<sdkmetrics::SketchAggregator<T>> temp_batch_agg_sketch =
+          std::dynamic_pointer_cast<sdkmetrics::SketchAggregator<T>>(batch_agg);
+
+      std::shared_ptr<sdkmetrics::SketchAggregator<T>> temp_record_agg_sketch =
+          std::dynamic_pointer_cast<sdkmetrics::SketchAggregator<T>>(record_agg);
+
+      auto temp_batch_agg_raw_sketch  = temp_batch_agg_sketch.get();
+      auto temp_record_agg_raw_sketch = temp_record_agg_sketch.get();
+
+      temp_batch_agg_raw_sketch->merge(*temp_record_agg_raw_sketch);
+    }
+    else if (agg_kind == sdkmetrics::AggregatorKind::Histogram)
+    {
+      std::shared_ptr<sdkmetrics::HistogramAggregator<T>> temp_batch_agg_histogram =
+          std::dynamic_pointer_cast<sdkmetrics::HistogramAggregator<T>>(batch_agg);
+
+      std::shared_ptr<sdkmetrics::HistogramAggregator<T>> temp_record_agg_histogram =
+          std::dynamic_pointer_cast<sdkmetrics::HistogramAggregator<T>>(record_agg);
+
+      auto temp_batch_agg_raw_histogram  = temp_batch_agg_histogram.get();
+      auto temp_record_agg_raw_histogram = temp_record_agg_histogram.get();
+
+      temp_batch_agg_raw_histogram->merge(*temp_record_agg_raw_histogram);
+    }
+    else if (agg_kind == sdkmetrics::AggregatorKind::Exact)
+    {
+      std::shared_ptr<sdkmetrics::ExactAggregator<T>> temp_batch_agg_exact =
+          std::dynamic_pointer_cast<sdkmetrics::ExactAggregator<T>>(batch_agg);
+
+      std::shared_ptr<sdkmetrics::ExactAggregator<T>> temp_record_agg_exact =
+          std::dynamic_pointer_cast<sdkmetrics::ExactAggregator<T>>(record_agg);
+
+      auto temp_batch_agg_raw_exact  = temp_batch_agg_exact.get();
+      auto temp_record_agg_raw_exact = temp_record_agg_exact.get();
+
+      temp_batch_agg_raw_exact->merge(*temp_record_agg_raw_exact);
+    }
+  }
+
+  std::map<metrics_api::InstrumentKind, std::string> ins_to_string{
+      {metrics_api::InstrumentKind::Counter, "Counter"},
```

Comment:
minor: consider using macro to simplify this (string tagging https://docs.microsoft.com/cpp/preprocessor/stringizing-operator-hash).

Reply:
I've never seen something like this, its a much better solution. Thanks!